capstone-sys/build.rs

```diff
@@ -148,6 +152,62 @@ fn env_var(var: &str) -> String {
     env::var(var).expect(&format!("Environment variable {} is not set", var))
 }
 
+/// Parse generated bindings and impl from_insn_id() for all architectures
+/// instructions enum declaration.
+#[cfg(feature = "use_bindgen")]
+fn impl_insid_to_insenum(bindings: &str) -> String {
+
+    let mut impl_arch_enum = String::new();
+
+    // find all architectures instructions enum declaration
+    let re = Regex::new(r"pub enum (?P<arch>.*)_insn (?s)\{.*?\}")
+        .expect("Unable to compile regex");
+
+    for cap in re.captures_iter(&bindings) {
+        impl_arch_enum.push_str(
+            &format!("impl {}_insn {{\n
+            pub fn from_insn_id(id: u32) -> {}_insn {{\n
+            match id {{\n",
+            &cap["arch"], &cap["arch"]));
+
+        // find instructions and their id
+        let re2 = Regex::new(
+            &format!("{}_INS_(?P<ins>[A-Z]+) = (?P<id>\\d+)",
```

Comment:
You need to account for underscores, numbers, and possible after `_INS_`.
We want to have this be future-proof.

For example, you are missing `MIPS_INS_ABSQ_S`, `MIPS_INS_ADDIUR1SP`.

Reference: https://doc.rust-lang.org/stable/reference/identifiers.html